client/src/pages/ProfilePage.jsx

```diff
@@ -0,0 +1,447 @@
+                <p className="help is-danger">{errors.password}</p>
+              ) : (
+                <p className="help">At least 6 characters</p>
+              )}
+            </div>
+
+            <div className="field">
+              <label htmlFor="confirm-password" className="label">
+                Confirm password
+              </label>
+
+              <div className="control has-icons-left has-icons-right">
+                <Field
+                  validate={validatePassword}
+                  name="confirmation"
+                  type="password"
+                  id="confirm-password"
+                  placeholder="*******"
+                  className={cn('input', {
+                    'is-danger': touched.confirmation && errors.confirmation,
+                  })}
+                />
+
+                <span className="icon is-small is-left">
+                  <i className="fa fa-lock"></i>
+                </span>
+
+                {touched.confirmation && errors.confirmation && (
+                  <span className="icon is-small is-right has-text-danger">
+                    <i className="fas fa-exclamation-triangle"></i>
+                  </span>
+                )}
+              </div>
+
+              {touched.confirmation && errors.confirmation ? (
+                <p className="help is-danger">{errors.confirmation}</p>
+              ) : (
+                <p className="help">At least 6 characters</p>
+              )}
+            </div>
+
+            <div className="field">
+              <button
+                type="submit"
+                className={cn('button is-success has-text-weight-bold', {
+                  'is-loading': isSubmitting,
+                })}
+                disabled={
+                  isSubmitting ||
+                  errors.oldPassword ||
+                  errors.password ||
+                  errors.confirmation
+                }
+              >
+                Submit
+              </button>
+            </div>
+          </Form>
+        )}
+      </Formik>
+
+      <Formik
+        initialValues={{
+          newEmail: '',
+          password: '',
+        }}
+        validateOnMount={true}
+        onSubmit={({ newEmail, password }, formikHelpers) => {
+          formikHelpers.setSubmitting(true);
+
+          userService
+            .changeEmail({ userId, newEmail, password })
+            .then((response) => {
+              alert(response);
+              checkAuth();
+              formikHelpers.resetForm();
+              navigate('/login');
+            })
+            .catch((error) => {
+              if (error.message) {
+                setError(error.message);
+              }
+
+              if (!error.response?.data) {
+                return;
+              }
+
+              const { errors, message } = error.response.data;
+
+              formikHelpers.setFieldError('email', errors?.newEmail);
```

Comment:
The field name 'email' used in setFieldError should be 'newEmail' to match the form field name.